### PR TITLE
fix retribert's `test_torch_encode_plus_sent_to_model`

### PR DESCRIPTION
**Pull request by SaulLu**
# What does this PR do?

This PR proposes a fix for the slow test `test_torch_encode_plus_sent_to_model` for RetriBert that failed in the CI daily that ran yesterday.

The error is due to the fact that RetriBert is not a classical model in the sense that the input can be fed to one or two encoders of the model. As a result `get_input_embeddings` is an unimplemented method and the forward method expects more arguments in input than what is outputed by the tokenizer.

I have therefore overridden the common test with a specific test for RetriBert that reflects these two specificities, which I will highlight them in a comment below.

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [x] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed

## Local test

I've tested it by running:
```bash
RUN_SLOW=yes pytest tests/models/retribert/ -k test_torch_encode_plus_sent_to_model
```
output:
```
=========================================================================================== test session starts ============================================================================================
platform linux -- Python 3.9.12, pytest-7.1.1, pluggy-1.0.0
rootdir: /home/lucile_huggingface_co/repos/transformers, configfile: setup.cfg
plugins: dash-2.3.1, hypothesis-6.41.0, timeout-2.1.0, forked-1.4.0, xdist-2.5.0
collected 98 items / 97 deselected / 1 selected                                                                                                                                                            

tests/models/retribert/test_tokenization_retribert.py .                                                                                                                                              [100%]

============================================================================== 1 passed, 97 deselected, 10 warnings in 5.38s ===============================================================================
```

### CODE REVIEW

File: tests/models/retribert/test_tokenization_retribert.py [hunk 1]
```diff
@@ -338,3 +338,47 @@ def test_change_tokenize_chinese_chars(self):
...
+                config_class, model_class = MODEL_TOKENIZER_MAPPING[tokenizer.__class__]
+                config = config_class()
+
+                if config.is_encoder_decoder or config.pad_token_id is None:
+                    return
+
+                model = model_class(config)
+
+                # The following test is different from the common's one
+                self.assertGreaterEqual(model.bert_query.get_input_embeddings().weight.shape[0], len(tokenizer))
```
Comment (SaulLu) [inline tests/models/retribert/test_tokenization_retribert.py:368]: In the common tests it was:
```python
                # Make sure the model contains at least the full vocabulary size in its embedding matrix
                is_using_common_embeddings = hasattr(model.get_input_embeddings(), "weight")
                if is_using_common_embeddings:
                    self.assertGreaterEqual(model.get_input_embeddings().weight.shape[0], len(tokenizer))
```

File: tests/models/retribert/test_tokenization_retribert.py [hunk 2]
```diff
@@ -338,3 +338,47 @@ def test_change_tokenize_chinese_chars(self):
...
+                # Ensure that the BatchEncoding.to() method works.
+                encoded_sequence.to(model.device)
+
+                batch_encoded_sequence = tokenizer.batch_encode_plus([sequence, sequence], return_tensors="pt")
+                # This should not fail
+
+                with torch.no_grad():  # saves some time
+                    # The following lines are different from the common's ones
+                    model.embed_questions(**encoded_sequence)
+                    model.embed_questions(**batch_encoded_sequence)
```
Comment (SaulLu) [inline tests/models/retribert/test_tokenization_retribert.py:384]: In the common test it was:
```python
                with torch.no_grad():  # saves some time
                    model(**encoded_sequence)
                    model(**batch_encoded_sequence)
```

**Review by ydshieh — APPROVED** [review]
LGTM, thank you @SaulLu .

Just a minor question: do you think it makes sense to test the same but for
```
self.bert_doc and embed_answers
``` 

I see
```
self.bert_query if self.bert_doc is None else self.bert_doc,
```
so probably it doesn't make sense in this test though.

**Comment by SaulLu** [comment]
@ydshieh , this is a good question (which I also asked myself, I should have shared my thoughts on it). 

Actually `self.bert_doc` is set to None in the model that is loaded. So, we can't test anything on `self.bert_doc` and `embed_answers` will be identical to `embed_questions` (and I remembered that we're testing the tokenizer not the model). But I follow what you think is best, i.e. if you think that it would be better to test the other 2 cases as a precaution if the checkpoint ever changes

**Comment by ydshieh** [comment]
> `self.bert_doc` is set to None

In this case, I think we can just keep what you have done so far in this PR, no need to change 😄 . Thanks for the info.

**Review by LysandreJik — APPROVED** [review]
Great, thank you for working on that @SaulLu!

**Comment by ydshieh** [comment]
@SaulLu Guess we can merge this PR now :-) ?

**Pull request merged**